In [ ]:
# bring the unknown deploy type values from the map in the FloodNet_Sensor_Location notebook
%store -r NaN_deploy_type_sensors

In [ ]:
# import libraries
import pandas as pd

In [ ]:
# read in the log of every deployed sensor in Directus
all_deployed_sensors = pd.read_csv('directus_all_deployments.csv')

In [ ]:
# view columns for clarity
print(NaN_deploy_type_sensors.columns)
print(all_deployed_sensors.columns)

Index(['sensor_name', 'deploy_type'], dtype='str')
Index(['deployment_id', 'dev_id', 'name', 'sensor_status', 'date_updated',
       'deploy_type', 'date_deployed', 'requested_by', 'date_down',
       'replaced_by_sensor'],
      dtype='str')


In [ ]:
# looking at how many of the NaN values are detected in all_deployed_sensors to diagnose issues
nan_in_deployed = all_deployed_sensors[all_deployed_sensors['name'].isin(NaN_deploy_type_sensors['sensor_name'])]
print(f"{len(nan_in_deployed)} of {len(NaN_deploy_type_sensors)} NaN-deploy-type sensors found in all_deployed_sensors")
nan_in_deployed


32 of 44 NaN-deploy-type sensors found in all_deployed_sensors


,deployment_id,dev_id,name,sensor_status,date_updated,deploy_type,date_deployed,requested_by,date_down,replaced_by_sensor
244,mind_deal_ensure,fs-00046,Q - 149th Ave/ Brookville Blvd,retired,2026-04-09T20:12:16.792Z,coastal,2023-06-22T00:00:00,"[""dep""]",2025-08-21T12:00:00,"[{""id"":12,""deployments_deployment_id"":{""slug"":..."
245,easily_dear_mouse,fs-00023,Q - Russell St 2,retired,2026-04-09T20:12:16.792Z,coastal,2021-03-05T00:00:00,"[""community""]",2024-02-15T00:00:00,"[{""id"":1,""deployments_deployment_id"":{""slug"":""..."
247,duly_star_bee,fs-00001,M - Morningside Av/125th St,retired,2026-04-09T20:12:16.792Z,pluvial,2021-12-12T00:00:00,[],2023-10-12T00:00:00,"[{""id"":6,""deployments_deployment_id"":{""slug"":""..."
248,hardly_keen_bird,fs3-00191,BK - Mermaid Ave/W 27th St,retired,2026-04-09T20:12:16.792Z,pluvial,2024-04-19T00:00:00,"[""community""]",2025-08-21T12:00:00,"[{""id"":13,""deployments_deployment_id"":{""slug"":..."
250,duly_wanted_akita,fs3-00134,Q - 96th St/55th Ave,retired,2026-04-09T20:12:16.792Z,pluvial,2024-08-14T13:00:00,"[""community""]",2024-10-14T17:10:00,[]
251,hugely-live-hermit,fs3-00164,BX - E 165th St/River Ave,retired,2026-04-09T20:12:16.792Z,pluvial,2024-08-22T11:35:00,"[""dep""]",2025-10-24T12:00:00,[]
252,daily_expert_whale,fs3-00013,BK - 45th St/5th Ave,retired,2026-04-09T20:12:16.792Z,pluvial,2024-04-18T00:00:00,"[""community ""]",2024-12-20T12:00:00,[]
253,humbly-flying-buck,fs3-00315,BX - Morris Ave/E 170th St,retired,2026-04-09T20:12:16.792Z,pluvial,2024-08-22T11:18:00,"[""dep""]",2025-10-24T12:00:00,[]
254,only_vast_lizard,fs3-00066,M - E 116th St/FDR Drive,retired,2026-04-09T20:12:16.792Z,pluvial,2024-07-03T15:33:00,"[""manhattan community board 11""]",2024-12-13T12:00:00,[]
255,overly_sharp_collie,fs3-00024,Q - 232nd St/147th Ave,retired,2026-04-09T20:12:16.792Z,pluvial,2024-05-16T00:00:00,"[""community""]",2024-07-25T12:00:00,[]


The NaN values in all deployed sensors are mostly retired. That's why they're coming up as NaN in the FloodNet_Sensor_Location maps. 

In [ ]:
# looking at the 12 sensors that weren't detected in all_deployed_sensors
not_in_deployed = NaN_deploy_type_sensors[~NaN_deploy_type_sensors['sensor_name'].isin(all_deployed_sensors['name'])]
print(f"{len(not_in_deployed)} sensors not found in all_deployed_sensors")
not_in_deployed


12 sensors not found in all_deployed_sensors


,sensor_name,deploy_type
0,BK - Georgia Ave/ Livonia Ave,NaN
37,BK - Ave V/ Stillwell Ave,NaN
43,BK - W 31st Ave/Neptune Ave,NaN
47,Q - Battery Rd/Chandler St,NaN
51,BX - Lincoln Ave / Bruckner Blvd,NaN
84,BK - Tillary St / Navy St,NaN
171,SI - Snug Harbor Rd / Kissel Ave,NaN
218,Q - Bell Blvd / Northern Blvd,NaN
267,M - Broad St/South St,NaN
270,BK - Coney Island Ave/Park Circle,NaN


FloodNet_Sensor_Locations has the above 12 undetected sensors listed in the dataframe labeled "unmatched" due to data cleaning issues. The sensors exist, but their names are different in some way when  The following is the list of these sensors as compared with their statuses on the Directus website:
- 0    BK - Georgia Ave/ Livonia Ave: status good, deploy type pluvial, deployed march 2026
- 37   BK - Ave V/ Stillwell Ave: status good, deploy type pluvial, deployed 4/15/26
- 43   BK - W 31st Ave/Neptune Ave: status dead, deploy type pluvial, deployed 3/12/26
- 47   Q - Battery Rd/Chandler St: status dead, deploy type coastal, deployed 3/27/26
- 51   BX - Lincoln Ave / Bruckner Blvd: status good, deploy type pluvial
- 171  SI - Snug Harbor Rd / Kissel Ave: status good, pluvial
- 218  Q - Bell Blvd / Northern Blvd: status noisy, pluvial
- 267  M - Broad St/South St: status good, pluvial
- 270  BK - Coney Island Ave/Park Circle: status good, pluvial
- 281  Q - Beach 43rd St: status good, pluvial
- 346  BK - Kingston Ave/Rutland Rd: status good, pluvial
